apps/lib/apps.c

```diff
@@ -197,7 +197,7 @@ int set_nameopt(const char *arg)
 
 unsigned long get_nameopt(void)
 {
-    return (nmflag_set) ? nmflag : XN_FLAG_ONELINE;
+    return (nmflag_set) ? nmflag : ASN1_STRFLGS_UTF8_CONVERT;
```

Comment:
Couldn't this just be `return nmflag;` and the `nmflag_set` variable removed completely?  It might need some logic cleanup in `set_nameopt()` above to never change the value on failure:

```c
int set_nameopt(const char *arg)
{
    unsigned long int tmp = nmflag;
    int ret = set_name_ex(&tmp, arg);

    if (ret)
        nmflag = tmp;
    return ret;
}
```

Reply:
I'm not sure the `tmp = nmflag` is required in my alternative.